base/src/main/java/proguard/optimize/kotlin/KotlinLambdaMerger.java

```diff
@@ -0,0 +1,440 @@
+        {
+            throw new IllegalArgumentException("Lambda class " + externalClassName + " cannot be merged, because it " +
+                                               "contains a static method that could be used outside the class itself.");
+        }
+        else if (!lambdaClassIsNotDirectlyInvoked(lambdaClass))
+        {
+            throw new IllegalArgumentException("Lambda class " + externalClassName + " cannot be merged, because it is " +
+                                               "directly invoked with its specific invoke method.");
+        }
+        else if (!nonINSTANCEFieldsAreNotReferencedFromSamePackage(lambdaClass, programClassPool))
+        {
+            throw new IllegalArgumentException("Lambda class " + externalClassName + " cannot be merged, because one of " +
+                                               "its fields, other than the 'INSTANCE' field is referenced by one of its " +
+                                               "inner classes.");
+        }
+        else if (!lambdaClassHasTotalMethodCodeSizeThatCanBeInlined(lambdaClass))
+        {
+            throw new IllegalArgumentException("Lambda class " + externalClassName + " cannot be merged, because its " +
+                                               "methods are too big to be inlined.");
+        }
+        else if (!mergeLambdaClassesWithUnexpectedMethods && !lambdaClassHasNoUnexpectedMethods(lambdaClass))
+        {
+            throw new IllegalArgumentException("Lambda class " + externalClassName + " cannot be merged, because it " +
+                                               "contains unexpected methods.");
+        }
+    }
+
+    private static boolean lambdaClassHasNoBootstrapMethod(ProgramClass lambdaClass) {
+        AttributeCounter attributeCounter = new AttributeCounter();
+        lambdaClass.attributeAccept(Attribute.BOOTSTRAP_METHODS, attributeCounter);
+        return attributeCounter.getCount() == 0;
+    }
+
+    private static boolean lambdaClassHasNoAccessibleStaticMethods(ProgramClass lambdaClass)
+    {
+        MethodCounter nonPrivateStaticMethodCounter = new MethodCounter();
+        String regularExpression = "!" + ClassConstants.METHOD_NAME_CLINIT;
+        lambdaClass.methodsAccept(new MemberAccessFilter(AccessConstants.STATIC, AccessConstants.PRIVATE,
+                                  new MemberNameFilter(regularExpression,
+                                  new MultiMemberVisitor(
+                                  nonPrivateStaticMethodCounter,
+                                  new MemberVisitor() {
+                                      @Override
+                                      public void visitProgramMethod(ProgramClass programClass,
+                                                                     ProgramMethod programMethod) {
+                                          logger.warn("Lambda class {} contains a static method that cannot be merged: {}",
```

Comment:
```suggestion
                                          logger.trace("Lambda class {} contains a static method that cannot be merged: {}",
```

Reply:
I don't think we need all these warnings, the user cannot do anything about it and nothing bad is happening only fewer lambdas are merged.